test/test_cpp_extensions_aot.py

```diff
@@ -8,10 +8,14 @@
 import torch.backends.cudnn
 import torch.utils.cpp_extension
 
+import pytest
```

Comment:
The `calculate_shards` part of `test/run_test.py` is clear as mud, but I think the tests are run. Probably `pytest` is installed on the relevant CI jobs though - it looks like this should be added to `USE_PYTEST_LIST`.

Reply:
What I mean is, the import may be missing currently, `pytest` is still an optional dependency today. In an env with no pytest installed:
```
$ python test/test_cpp_extensions_aot.py 
...
Traceback (most recent call last):
  File "test/test_cpp_extensions_aot.py", line 11, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

Install `pytest`, try again:
```
$ python test/test_cpp_extensions_aot.py 
...
Traceback (most recent call last):
  File "test/test_cpp_extensions_aot.py", line 14, in <module>
    import torch_test_cpp_extension.cpp as cpp_extension
ModuleNotFoundError: No module named 'torch_test_cpp_extension'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "test/test_cpp_extensions_aot.py", line 21, in <module>
    ) from e
RuntimeError: test_cpp_extensions_aot.py cannot be invoked directly. Run `python run_test.py -i test_cpp_extensions_aot_ninja` instead
```

So at least it may hide the informative error message.

Reply:
Oh! I thought we had agreed we can make it mandatory haha. OK, I can probably guard this a bit more.

Reply:
Yes we can! And will! It's just not done yet.